SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -500,24 +546,39 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 		return;
 	}
 
-	// If we're the client, we can now ask the server to spawn our controller.
-	if (!IsServer())
+	if (IsServer())
 	{
-		// If we know the GSM is already accepting players, simply spawn.
-		if (GlobalStateManager->bAcceptingPlayers && GetWorld()->RemovePIEPrefix(GlobalStateManager->DeploymentMapURL) == GetWorld()->RemovePIEPrefix(GetWorld()->URL.Map))
+		if (GlobalStateManager != nullptr)
 		{
-			PlayerSpawner->SendPlayerSpawnRequest();
-			bWaitingForAcceptingPlayersToSpawn = false;
+			// Increment the session id, so users don't rejoin the old game.
+			GlobalStateManager->SetCanBeginPlay(true);
+			GlobalStateManager->TriggerBeginPlay();
+			GlobalStateManager->SetAcceptingPlayers(true);
+			GlobalStateManager->IncrementSessionID();
 		}
 		else
 		{
-			checkNoEntry();
+			UE_LOG(LogSpatial, Log, TEXT("Trying to call GlobalStateManager->SetAcceptingPlayers(true) when GlobalStateManager is nullptr"),);
 		}
 	}
+	else if (CanSendPlayerSpawnRequests())
+	{
+		MakePlayerSpawnRequest();
+	}
```

Comment:
I think clients will lock on a screen if they ever load the wrong map and thus cannot spawn. I don't expect it to happen due to the fact clients get the map to load by `QueryGSMToLoadMap` but in-case there is an edge case where a client queries the GSM and then server travel begins, client finishes loading map but it's now on the wrong one because the server is changing map...
Let's add a warning here and then trigger `QueryGSMToLoadMap` again here.